WalletWasabi.Gui/Global.cs

```diff
@@ -247,6 +247,15 @@ public async Task InitializeNoWalletAsync(TerminateService terminateService)
 					BitcoinStore.BlockRepository);
 
 				WalletManager.RegisterServices(BitcoinStore, Synchronizer, Config.ServiceConfiguration, HostedServices.Get<HybridFeeProvider>(), blockProvider);
+
+				try
+				{
+					UiConfig.RunOnSystemStartup = await StartupChecker.GetCurrentValueAsync().ConfigureAwait(false);
+				}
+				catch (Exception e)
+				{
+					Logger.LogWarning($"Failed to validate the start up option, error: {e.Message}.");
```

Comment:
```suggestion
					Logger.LogWarning($"Failed to get the start up setting, error: {e.Message}.");
```
?

Also, it should be more like

```suggestion
					Logger.LogWarning($"Failed to get the start up setting.", e);
```

which works if you add to `Logger.cs` the following piece of code:

```csharp
/// <summary>
/// Logs <paramref name="message"/> with <paramref name="exception"/> using <see cref="Exception.ToString()"/> concatenated to it at <see cref="LogLevel.Warning"/> level.
///
/// <para>For abnormal or unexpected events in the application flow.</para>
/// </summary>
/// <remarks>
/// <para>Includes situations when errors or other conditions occur that do not cause the application to stop, but which may need to be investigated.</para>
/// <para>Handled exceptions are a common place to use the <see cref="LogLevel.Warning"/> log level.</para>
/// </remarks>
/// <example>For example: <c>FileNotFoundException for file quotes.txt.</c></example>
public static void LogWarning(string message, Exception exception, [CallerFilePath] string callerFilePath = "", [CallerMemberName] string callerMemberName = "", [CallerLineNumber] int callerLineNumber = -1)
	=> Log(message, exception, LogLevel.Warning, callerFilePath: callerFilePath, callerMemberName: callerMemberName, callerLineNumber: callerLineNumber);
```